### PR TITLE
fix: underscore crate name

### LINKED ISSUE #37: Does discovery account for dash / underscore difference in path vs module name?
I have a module, part of a workspace: this is its `Cargo.toml`
```toml
[package]
name = "child-mod"
```

It's a submodule of `current-crate` where I'm generating Openapi docs.

```
current-crate/
├─ src/
│  ├─ mod.rs
│  ├─ child-mod/
│  │  ├─ src/
│  │  │  ├─ models.rs
```

Hence, I need to both declare 
- current crate (because it's part of the workspace) - that part works fine ✅
- the submodule with `from` keyword, as specified [here](https://docs.rs/crate/utoipauto/latest), like this:

```
#[utoipauto(paths = "./current-crate/src,./src/models.rs from child-mod")]`
```

Problem is, importing from submodule looks like this (note `_` instead of `-`):

```rust
use child_mod::SomeStruct;
```

So you receive a file path in the macro and [transform it to module path](https://github.com/ProbablyClem/utoipauto/blob/main/utoipauto-core/src/file_utils.rs#L70). 

Also, you [recurse through the file tree](https://github.com/ProbablyClem/utoipauto/blob/main/utoipauto-core/src/discover.rs#L11) so I cannot directly input underscored file name in the `#[utoipauto(path` annotation because the file wouldn't be found.

However I couldn't find whether you account for this module naming quirk: difference between file paths containing `-` and module paths containing `_`. 

Is this why my `path` annotation isn't working or something else?

### PR DESCRIPTION
**Pull request by EvolveArt**
Resolves #37 

## Changes
- Adds the `replace` call as mentioned in the issue's comments
- Includes corresponding unit and acceptance test

### CODE REVIEW

**Comment by DenuxPlays** [comment]
Does normal crates name liked `normal-crate` still work?
I am not sure if this is tested?

**Review by DenuxPlays — CHANGES_REQUESTED** [review]
Unit test looks fine.
But I am missing an acceptance test.
I think we already have a test crate in the tests folder which could be extended.

**Comment by DenuxPlays** [comment]
> > Unit test looks fine. But I am missing an acceptance test. I think we already have a test crate in the tests folder which could be extended.
> 
> 
> 
> Should I create another `create-segment-path` for instance ?  or just a `sub-folder` ?
> 
> 
> 
> EDIT: the issue is on the crate name, so the only acceptance test that could be done is by creating a new test crate
> 
> EDIT 2: the tests folder doesn't seem to be compiled anywhere

Yes just create another test crate.

I noticed this too so I already created a fix: https://github.com/ProbablyClem/utoipauto/pull/39

**Comment by EvolveArt** [comment]
> > > Unit test looks fine. But I am missing an acceptance test. I think we already have a test crate in the tests folder which could be extended.
> > 
> > 
> > Should I create another `create-segment-path` for instance ?  or just a `sub-folder` ?
> > EDIT: the issue is on the crate name, so the only acceptance test that could be done is by creating a new test crate
> > EDIT 2: the tests folder doesn't seem to be compiled anywhere
> 
> Yes just create another test crate.
> 
> I noticed this too so I already created a fix: #39

Ok so I will rebase once your fix is merged

**Comment by DenuxPlays** [comment]
@EvolveArt 

PRs have been merged you can now merge main into your branch and create the acceptance test

**Comment by EvolveArt** [comment]
> @EvolveArt
> 
> PRs have been merged you can now merge main into your branch and create the acceptance test

Done, you can check. Not sure if that's the best way but it does provide a correct acceptance test. 
I think acceptance tests could be refactored in another PR.

File: acceptance/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [workspace]
-members = ["crate_segment_path"]
+members = ["crate_segment_path", "new_crate_segment_path"]
```
Comment (DenuxPlays) [inline acceptance/Cargo.toml:2]:
```suggestion
members = ["crate_segment_path", "folder_in_src"]
```

File: acceptance/crate_segment_path/Cargo.toml [hunk 2]
```diff
@@ -15,4 +15,5 @@ path = "crate_folder/lib.rs"
 
 [dependencies]
 utoipa.workspace = true
-utoipauto.workspace = true
\ No newline at end of file
+utoipauto.workspace = true
+utoipauto-lib-test-new = { path = "../new_crate_segment_path" }
```
Comment (DenuxPlays) [inline acceptance/crate_segment_path/Cargo.toml:19]:
```suggestion
folder_in_src = { path = "../folder_in_src" }
```

File: acceptance/new_crate_segment_path/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "utoipauto-lib-test-new"
```
Comment (DenuxPlays) [inline acceptance/new_crate_segment_path/Cargo.toml:2]:
```suggestion
name = "folder_in_src"
```

**Review by DenuxPlays — CHANGES_REQUESTED** [review]
Please rename the folder and the name to `folder_in_src`
The name is more descriptive.

Otherwise your pr looks pretty good.
Thank you for your work :)

**Comment by EvolveArt** [comment]
> Please rename the folder and the name to `folder_in_src` The name is more descriptive.
> 
> Otherwise your pr looks pretty good. Thank you for your work :)

Done :) looking forward to help more, this crate has been really useful to simplify our utoipa doc generation

**Review by DenuxPlays — APPROVED** [review]
Looks good to me

**Comment by DenuxPlays** [comment]
Note:

I am not sure if we should introduce a "main" testing crate where we could merge all things that need to be tested across multiple crates (like this pr).
But this has nothing todo with this pr.
Just a small note from me.

Note 2:

I do have a plan for creating a larger test crate that tests multiple things maybe we can combine the ideas.

**Comment by DenuxPlays** [comment]
@EvolveArt Could you rebase your pr one more time?
I fixed the ci a bit

**Comment by EvolveArt** [comment]
> @EvolveArt Could you rebase your pr one more time? I fixed the ci a bit

done

**Review by DenuxPlays — APPROVED** [review]

**Comment by ProbablyClem** [comment]
Thanks guy for your work

**Pull request merged**